### DISCUSSION TITLE
**Urgent Help**

### DISCUSSION THREAD

**Discussion by Naveen-Crowdera** [vercel/next.js#35812] [Help]
PS C:\crowdera-platform> npm run build

> crowdera-platform@2.1.0 build
> next build

info  - Loaded env from C:\crowdera-platform\.env
info  - Using webpack 5. Reason: no custom webpack configuration in next.config.js https://nextjs.org/docs/messages/webpack5
info  - Checking validity of types
info  - Creating an optimized production build
Failed to compile.

ModuleNotFoundError: Module not found: Error: Empty dependency (no request)

> Build error occurred
Error: > Build failed because of webpack errors
    at C:\crowdera-platform\node_modules\next\dist\build\index.js:17:924
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Span.traceAsyncFn (C:\crowdera-platform\node_modules\next\dist\telemetry\trace\trace.js:6:584)

**Comment by icyJoseph** [comment]
Hi,

We need more info. 

I am gonna guess that you are importing a module that can't be bundled into the client, or that you have not installed dependencies.

I hope you understand that the information you have given is not enough to help you out.

- Next version
- What are you trying to build
- What dependencies are involved
- https://next-code-elimination.vercel.app/ use this tool to verify that the client bundle is as expected
- etc, etc...

**Reply by Naveen-Crowdera** [reply]
"dependencies": {
    "@fortawesome/fontawesome-free": "5.15.3",
    "@heroicons/react": "^1.0.0",
    "@material-ui/core": "^4.12.3",
    "@material-ui/data-grid": "^4.0.0-alpha.24",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/lab": "^4.0.0-alpha.60",
    "@material-ui/x-grid-data-generator": "^4.0.0-alpha.24",
    "@material-ui/x-license": "^4.0.0-alpha.36",
    "@popperjs/core": "2.9.1",
    "@reach/menu-button": "^0.16.1",
    "@reduxjs/toolkit": "^1.8.1",
    "@stripe/react-stripe-js": "^1.4.0",
    "@stripe/stripe-js": "^1.13.2",
    "@tailwindcss/forms": "0.2.1",
    "@types/node": "^17.0.21",
    "@types/react": "^17.0.41",
    "amplitude-js": "^8.5.0",
    "axios": "^0.21.1",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "chart.js": "2.9.4",
    "currency-symbol-map": "^5.0.1",
    "disqus-react": "^1.0.11",
    "dompurify": "^2.2.7",
    "draft-js": "^0.11.7",
    "draftjs-to-html": "^0.9.1",
    "firebase": "^8.3.2",
    "firebase-admin": "^9.6.0",
    "formik": "^2.2.6",
    "gulp": "4.0.2",
    "gulp-append-prepend": "1.0.8",
    "html-to-draftjs": "^1.5.0",
    "material-ui-chip-input": "^2.0.0-beta.2",
    "material-ui-image": "^3.3.2",
    "moment": "^2.29.1",
    "moment-timezone": "^0.5.33",
    "next": "^10.0.9",
    "next-seo": "^4.24.0",
    "next-useragent": "^2.6.0",
    "node": "^16.1.0",
    "nookies": "^2.5.2",
    "pm2": "^5.1.0",
    "prop-types": "^15.7.2",
    "react": "17.0.1",
    "react-alice-carousel": "^2.4.0",
    "react-cookie-consent": "^6.2.3",
    "react-credit-card-input": "^1.1.5",
    "react-credit-cards": "^0.8.3",
    "react-dom": "17.0.1",
    "react-draft-wysiwyg": "^1.14.7",
    "react-dropzone": "^11.3.2",
    "react-facebook": "^8.1.4",
    "react-firebaseui": "^4.1.0",
    "react-froala-wysiwyg": "^3.2.7",
    "react-google-charts": "^3.0.15",
    "react-hook-form": "^6.15.5",
    "react-hotjar": "^3.0.1",
    "react-html-parser": "^2.0.2",
    "react-infinite-scroll-component": "^6.1.0",
    "react-moment": "^1.1.1",
    "react-phone-input-2": "^2.14.0",
    "react-redux": "^7.2.3",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "react-select": "^4.3.0",
    "react-share": "^4.4.0",
    "react-show-more-text": "^1.4.6",
    "react-stepper-horizontal": "^1.0.11",
    "react-swipeable-views": "^0.13.9",
    "reactour": "^1.18.4",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "sass": "^1.32.12",
    "styled-components": "^5.3.0",
    "sweetalert2": "^10.15.7",
    "sweetalert2-react-content": "^3.3.1",
    "swiper": "^6.5.9",
    "tailwindcss": "2.0.4"
  },
  "scripts": {
    "start": "next start",
    "build": "next build",
    "dev": "next dev",
    "install:clean": "rm -rf node_modules/ && rm -rf package-lock.json && npm install && npm run build:tailwind && npm run dev",
    "build:tailwind": "tailwind build styles/index.css -o styles/tailwind.css"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "optionalDependencies": {
    "typescript": "^4.2.3"
  },
  "devDependencies": {
    "autoprefixer": "10.2.5",
    "postcss": "8.2.8"
  },

  "browser": {
    "child_process": false
  }

**Reply by Naveen-Crowdera** [reply]
before deploying we are check run build, here i'm getting this issue

**Reply by icyJoseph** [reply]
Still not enough info, but much better. My guess:

- `child_process: false`, I think the problem might be here. You need to reorder your imports, so that `child_process` is not needed by the client, at all, because it just doesn't exist on the client, and that you can build without this hack.

A couple of additional things there:

- Using ~ or ^ to mark dependencies could already be problematic. Not everyone respects semver. I think this is not the issue.
- You have `react-scripts` also installed
- Although, for web application, everything is really a devDependency, it is good to divide these into, those that will end up in the web application `dependencies`, and the rest `devDependencies`. 


What node version are you using? And is it possible for you to organise your imports/exports so that `child_process` is no longer needed? Or is that needed because of a library you use?

**Reply by Naveen-Crowdera** [reply]
shill i remove this child_process: false  and reordering  imports means??, i'm not getting how to reorder it

**Reply by icyJoseph** [reply]
Well, without access to your repository I can't tell you either, and this is a guess.

As I see it, the problem is that, `child_process: false` should not be necessary, there must be a module (yours or from node_modules) that's using child_process, and someone that's leaking into the client side code. 

Was this working before? what were you trying to do? There's still a lot of information missing, but I'd start by removing the `child_process: false` and see if it builds, or if there's a different error message.

**Reply by Naveen-Crowdera** [reply]
sorry.. i can't give access because its my professional Repo, and i tried removing child_process: false, but end up in same issue

**Reply by Naveen-Crowdera** [reply]
yeah before it was working

**Reply by Naveen-Crowdera** [reply]
we trying to deploy the code where in vercel  we found issues and  i'm trying build in my local and checking

**Reply by Naveen-Crowdera** [reply]
hey, any update regards help

**Reply by icyJoseph** [reply]
Hi, well without access to the repository, or more error logs, and with over 30 dependencies, there's very little we could do to help out.

- Do you have any special webpack configuration? 
- I'd suggest, to create a new folder, `old-pages`, and move all of your pages to `old-pages`,  and then ONE by ONE, move them back to `pages`, and build, eventually one of them has to break the build, and from there you have to see which imports are used by the page. Again, try to hunt down the dependency that's breaking the build by removing them one by one. DO NOT COMMIT THIS PROCEDURE.

**Comment by Naveen-Crowdera** [comment]
hey joseph  tq i've tried the same method, we found the bug which  creating the issue
